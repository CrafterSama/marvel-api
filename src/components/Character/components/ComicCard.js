import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { Card, Overlayer, CharacterInfo, CharacterName } from '../../../styles'

const ComicCard = ({ data }) => {

  return (
    <Card background={data.thumbnail}>
      <Overlayer />
      <Link to={`comics/${data.id}`}>
        <CharacterInfo>
          <CharacterName>
            <span>{data.name}</span>
          </CharacterName>
          <div className='character-favorite'></div>
        </CharacterInfo>
      </Link>
    </Card>
  );
};

ComicCard.propTypes = {
  data: PropTypes.object.isRequired
};

export default ComicCard;
