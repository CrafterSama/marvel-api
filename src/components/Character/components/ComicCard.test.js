import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import ComicCard from './ComicCard';

configure({ adapter: new Adapter() });

describe('Test for <ComicCard /> Component', () => {


  test('matches snapshot', () => {
    const wrapper = shallow(
      <ComicCard
        data={ { data: {} } }
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
