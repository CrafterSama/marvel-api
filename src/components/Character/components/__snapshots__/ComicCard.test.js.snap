// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <ComicCard /> Component matches snapshot 1`] = `
<styles__Card>
  <styles__Overlayer />
  <Link
    to="comics/undefined"
  >
    <Component>
      <Component>
        <span />
      </Component>
      <div
        className="character-favorite"
      />
    </Component>
  </Link>
</styles__Card>
`;
